tests/test_match.py

```diff
@@ -280,3 +280,53 @@ def test_png_with_16_bits_per_channel():
     assert stbt.match(
         "tests/uint16.png",
         frame=cv2.imread(_find_file("uint8.png")))
+
+
+@requires_opencv_3
+def test_match_fast_path():
+    # This is just an example of typical use
+    assert stbt.match("action-panel-prototype.png",
+                      frame=stbt.load_image("action-panel.png"))
+
+
+@requires_opencv_3
+def test_that_match_fast_path_is_equivalent():
+    from _stbt.match import _load_image
+    black_reference = black(10, 10)
+    almost_black_reference = black(10, 10, value=1)
+    black_frame = black(1280, 720)
+    almost_black_frame = black(1280, 720, value=2)
+
+    images = [
+        ("videotestsrc-redblue.png", "videotestsrc-full-frame.png"),
+        ("action-panel.png", "action-panel.png"),
+        ("videotestsrc-full-frame.png", "videotestsrc-full-frame.png"),
+        ("videotestsrc-redblue-flipped.png", "videotestsrc-full-frame.png"),
+        ("button.png", "black-full-frame.png"),
+        ("completely-transparent.png", "buttons-on-blue-background.png"),
+        ("action-panel-template.png", "action-panel.png"),
+        ("button.png", "buttons.png"),
+        (black_reference, black_frame),
+        (almost_black_reference, black_frame),
+        (almost_black_reference, almost_black_frame),
+        ("repeating-pattern.png", "repeating-pattern-full-frame.png"),
+        ("button-transparent.png", "buttons.png"),
+    ]
+    for reference, frame in images:
+        if isinstance(frame, (str, unicode)):
+            frame = stbt.load_image(frame)
+        reference = _load_image(reference)
+        orig_m = stbt.match(reference, frame=frame)
+        fast_m = stbt.match(reference, frame=frame, region=orig_m.region)
+        assert orig_m.time == fast_m.time
+        assert orig_m.match == fast_m.match
+        assert orig_m.region == fast_m.region
+        assert bool(orig_m) == bool(fast_m)
+        if orig_m:
+            assert orig_m.first_pass_result == pytest.approx(
+                fast_m.first_pass_result, abs=0.1)
```

Comment:
0.1 seems quite high. What's the maximum difference you saw across the different implementations?

Reply:
This is caused by the pyramid levels.  With pyramid levels disabled they match closely.  In this case `if orig_m` implies that we reached the bottom pyramid level.  Without `if orig_m` and and with `abs=0.01` we get a failure: `0.833 != 0.808` (a difference of 0.025).

I'll change the test to vary its strictness depending on whether it matched or not.